### ISSUE TITLE
Cleaning cdk.out

### ISSUE THREAD

**Issue by vertti** [aws/aws-cdk#3749]
### General Information

Each cdk build creates a new assets folder under `cdk.out`. If this includes `node_modules`, the total size of the `cdk.out` folder can add up pretty quickly (mine was over 10Gb)!

### Description

I would like that the `cdk.out` folder would be kept in check.

### Proposed Solution

Some options that I briefly thought about:
- CDK could warn when the size of the `cdk.out` is over X Gb
- CDK could delete the whole `cdk.out` contents before starting to build new assets
- CDK could clean out assets older than X days
- Some configurable combination of the above

### Environment

  - **CDK CLI Version:** 1.4.0 (build 175471f)
  - **OS:** all
  - **Language:** Typescript (all?)

**Comment by RomainMuller** [comment]
This is a duplicate of #2869 -- closing.
